docs/features/environment-overrides.mdx

```diff
@@ -0,0 +1,431 @@
+---
+title: Environment-Level Overrides
+description: Dynamically control remote dependencies and environment variables per deployment environment
+---
+
+# Environment-Level Overrides
+
+One of Zephyr's most powerful capabilities is the ability to override application configuration on a per-deployment-environment basis. This means the same built application can dynamically adapt its behavior - consuming different versions of remote dependencies and using different environment variable values - depending on which environment URL serves it.
+
+## Overview
+
+Environment-level overrides enable two key capabilities:
+
+- **[Remote Dependency Overrides](#remote-dependency-overrides)**: Control which version, tag, or environment of remote modules your application loads
+- **[Environment Variable Overrides](#environment-variable-overrides)**: Dynamically inject environment-specific configuration values
+
+Both features work through the same underlying system: Zephyr captures configuration at build time in a manifest, then dynamically applies environment-specific overrides at runtime based on which environment serves your application.
+
+## The Manifest System
+
+Behind the scenes, Zephyr generates a `zephyr-manifest.json` file at build time that records:
+
+- Remote dependencies declared in your `zephyr:dependencies` configuration
+- Environment variables prefixed with `ZE_PUBLIC_` from your build environment
+
+This manifest enables runtime override capabilities - you don't need to interact with this file directly, but it's what makes environment-level customization possible without rebuilding your application.
+
+**Key benefit**: Build once, deploy everywhere with environment-specific configuration.
+
+## Remote Dependency Overrides
+
+When you build your application, Zephyr resolves your remote dependencies based on the version selectors in your `zephyr:dependencies` configuration. However, at the environment level, you can override these declarations to point to different versions, tags, or environments of your remotes without rebuilding.
+
+### Understanding Remote Overrides
+
+**Build-time**: Your application is built with remote dependency declarations from `package.json`:
+
+```json title="package.json"
+{
+  "zephyr:dependencies": {
+    "header": "ui-components@stable",
+    "cart": "cart-service@^2.0.0"
```

Comment:
Maybe we should keep the same name the remotes have to their aliases to avoid confustion.
And I mean this all across the examples.